### PR TITLE
Unify fusedev and virtiofs frontend

### PR DESCRIPTION
**Pull request by uran0sH**
We add "--run-mode" to specify the run mode such as virtiofs or
fusedev.

Signed-off-by: uran0sH <huangwenyuu@outlook.com>

### CODE REVIEW

**Comment by jiangliu** [comment]
We already have subcommand "fuse", "virtiofs" and "daemon".
Could we extend subcommand "daemon" to support all of "fuse", "virtiofs" and "fscache"?

**Comment by hsiangkao** [comment]
> We already have subcommand "fuse", "virtiofs" and "daemon". Could we extend subcommand "daemon" to support all of "fuse", "virtiofs" and "fscache"?

Yes, let's use subcommand `daemon` as a start

**Comment by imeoer** [comment]
@uran0sH Any updates for this?

**Comment by uran0sH** [comment]
I have added virtiofs and fuse to the daemon. And then what should I do?

**Comment by imeoer** [comment]
> I have added virtiofs and fuse to the daemon. And then what should I do?

Please take a look at the failed smoke test.

**Comment by changweige** [comment]
Please rebase this PR.
We should also remove fusedev and virtiofs related targets in Makefile.

**Comment by jiangliu** [comment]
The vhost-user feature should be disabled for macos, which may solve the failure of CI/macos-ut.

**Comment by imeoer** [comment]
We need to ensure image-service CI pass first, and then try to make openanolis CI pass by modifying the binary path here:
https://gitee.com/anolis/tone-cli/blob/master/tests/nydus-fscache-citest/run.sh#L54

File: Cargo.toml [hunk 1]
```diff
@@ -71,8 +71,7 @@ env_logger = "0.8.2"
 rand = "0.8.5"
 
 [features]
-fusedev = ["fuse-backend-rs/fusedev"]
-virtiofs = ["fuse-backend-rs/vhost-user-fs", "vm-memory", "vhost", "vhost-user-backend", "virtio-queue", "virtio-bindings"]
+default = ["fuse-backend-rs/fusedev", "fuse-backend-rs/vhost-user-fs", "vm-memory", "vhost", "vhost-user-backend", "virtio-queue", "virtio-bindings"]
```
Comment (jiangliu) [inline Cargo.toml:74]: We can't change the default feature this way. Due to limitation of cargo.toml, it will break macos builds. We could achieve the same by changing Makefile instead.
'''
default=["fusedev"]
fusedev = ["fuse-backend-rs/fusedev"]
virtiofs = ["fuse-backend-rs/vhost-user-fs", "vm-memory", "vhost", "vhost-user-backend", "virtio-queue", "virtio-bindings"]
'''

File: contrib/nydusify/pkg/checker/tool/nydusd.go [hunk 2]
```diff
@@ -162,6 +162,7 @@ func (nydusd *Nydusd) Mount() error {
 	nydusd.Umount(true)
 
 	args := []string{
+		"fuse",
```
Comment (jiangliu) [inline contrib/nydusify/pkg/checker/tool/nydusd.go:165]: Please revert this change, we are switching to subcommand form and deprecating new subcommand form

**Comment by jiangliu** [comment]
@uran0sH thanks for you great contribution:)
I have done some enhancements based on your version and post a new PR at https://github.com/dragonflyoss/image-service/pull/743
So we will close this one:)
